source/vistas/core/graphics/mesh.py

```diff
@@ -79,6 +79,13 @@ def __init__(
             glEnableVertexAttribArray(1)
             glVertexAttribPointer(1, 3, GL_FLOAT, GL_FALSE, sizeof(GLfloat) * 3, None)
 
+        if self.has_color_array:
```

Comment:
Yeah, that's what I was getting at; so that we were't doing iteration dependent on the dataset size in the application code.

At any rate, also out of scope for this PR; I think it's fine as-is for now. Mostly just wondering it it's possible and if it would make sense.